hack/ginkgo-e2e.sh

```diff
@@ -99,6 +99,11 @@ elif [[ "${KUBERNETES_PROVIDER}" == "aws" ]]; then
   auth_config=(
     "--auth_config=${HOME}/.kubernetes_auth"
   )
+elif [[ "${KUBERNETES_PROVIDER}" == "conformance_test" ]]; then
```

Comment:
This is kind of odd because it's not really telling the test framework anything about the provider. We later make inferences based on the provider to help neuter tests. We shouldn't, arguably, and should be inferring this based on other things, but I'm trying to figure out what the right pattern is here.

cc @jlowdermilk .. do you have any thoughts on how to get the config in here more better?